File: cp/codeforces/B_Gorilla_and_the_Exam.cpp
#include <iostream>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

int main() {
    int test;
    cin >> test;

    for (int t = 0; t < test; ++t) {
        int n, k;
        cin >> n >> k;
        
        vector<int> arr(n);
        for (int i = 0; i < n; ++i) {
            cin >> arr[i];
        }

        map<int, int> memo;
        for (int a : arr) {
            memo[a]++;
        }

        vector<int> brr;
        for (auto& pair : memo) {
            brr.push_back(pair.second);
        }

        sort(brr.begin(), brr.end());

        int curr_sum = 0;
        int ans = brr.size();

        for (int index = 0; index < brr.size(); ++index) {
            int curr = brr[index];

            if (k >= curr && index != n - 1) {
                ans--;
                k -= curr;
            } else {
                break;
            }
        }

        cout << max(ans, 1) << endl;
    }

    return 0;
}

File: cp/codeforces/B_Gorilla_and_the_Exam.py
from collections import defaultdict


test = int(input())

for _ in range(test):
    n, k = list(map(int, input().split()))
    arr = list(map(int, input().split()))
    memo = defaultdict(int)
    
    for a in arr:
        memo[a] += 1
    curr_sum = 0
    brr = sorted(memo.values())
    ans = len(brr)
    
    for index , curr in enumerate(brr):
        
        if k >= curr and index != n - 1:
            ans -= 1
            k -= curr
        else:
            break
        
            
    print(max(ans, 1))
            
    

